### PR TITLE
Use NSwag for OpenApi and typescript generation

### PR DESCRIPTION
**Pull request by raix**
### Summary & Motivation

Remove Swashbuckle in favor of NSwag. Swashbuckle requires .NET 7 sdk installed and development seem to be stale at the moment.

### Checklist

- [x] I have added a Label to the pull-request
- [x] I have added tests, and done manual regression tests
- [x] I have updated the documentation, if necessary

### CODE REVIEW

**Comment by tjementum** [comment]
FYI... Title should be a sentence without the issue number. Just making this comment to block merge ;)

File: application/shared-kernel/ApiCore/OpenApi/OpenApiDocumentGeneratorSettingsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+{
+    /// <summary>
+    ///     Configures the schema settings for the OpenAPI document generator
+    /// </summary>
+    public static void ConfigureSchemaSettings(
+        this AspNetCoreOpenApiDocumentGeneratorSettings settings,
+        IServiceProvider serviceProvider
+    )
+    {
+        if (settings.SchemaSettings is not SystemTextJsonSchemaGeneratorSettings schemaSettings) return;
```
Comment (tjementum) [inline application/shared-kernel/ApiCore/OpenApi/OpenApiDocumentGeneratorSettingsExtensions.cs:20]: I would inline this entire method, and then delete this check, as settings.SchemaSettings would then always be the same type.

File: application/shared-kernel/ApiCore/ApiCoreConfiguration.cs [hunk 2]
```diff
@@ -95,8 +95,8 @@ public static WebApplication AddApiCoreConfiguration<TDbContext>(this WebApplica
         app.MapDefaultEndpoints();
 
         // Enable Swagger UI
-        app.UseSwagger();
-        app.UseSwaggerUI(c => c.SwaggerEndpoint("/swagger/v1/swagger.json", "API"));
+        app.UseOpenApi();
+        app.UseSwaggerUi(); // Alternative: "app.UseReDoc();"
```
Comment (tjementum) [inline application/shared-kernel/ApiCore/ApiCoreConfiguration.cs:99]: Remove comment. Such discussions should not be in code.

File: application/shared-kernel/InfrastructureCore/InfrastructureCoreConfiguration.cs [hunk 3]
```diff
@@ -11,9 +11,7 @@ public static class InfrastructureCoreConfiguration
 {
     private static string? _cachedConnectionString;
 
-    public static readonly bool SwaggerGenerator = File.Exists(
-        Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!, "SwaggerGenerator.tmp")
-    );
+    public static readonly bool SwaggerGenerator = Environment.GetEnvironmentVariable("SWAGGER_GENERATOR") == "true";
```
Comment (tjementum) [inline application/shared-kernel/InfrastructureCore/InfrastructureCoreConfiguration.cs:14]: Does this work on Windows? I was not able to configure environment variables just for one build. That why I've added the hack with the `SwaggerGenerator.tmp` temp file.

File: application/shared-kernel/ApiCore/OpenApi/OpenApiDocumentGeneratorSettingsExtensions.cs (hunk 1 above)
Comment (raix) [reply application/shared-kernel/ApiCore/OpenApi/OpenApiDocumentGeneratorSettingsExtensions.cs:20]: Seems like this check is required as SchemaSettings is type `JsonSchemaGeneratorSettings`

File: application/account-management/Api/Api.csproj [hunk 4]
```diff
@@ -36,16 +40,14 @@
...
-        <OnError ExecuteTargets="CreatingSwaggerJson"/>
-    </Target>
-
-    <Target Name="CleanupSwaggerGeneratorTmp">
-        <Delete Files="$(OutputPath)SwaggerGenerator.tmp"/>
+    <PropertyGroup>
+         <RunPostBuildEvent>OnBuildSuccess</RunPostBuildEvent>
+     </PropertyGroup>
+
+    <Target Name="NSwag" AfterTargets="PostBuildEvent" Condition=" '$(Configuration)' == 'Debug' ">
```
Comment (tjementum) [inline application/account-management/Api/Api.csproj:47]: Why is this only for `Debug` builds. Is it not important that the `swagger.json` is also generated server side. Else we risk that it does not reflect the last-minute changes.

File: application/shared-kernel/InfrastructureCore/InfrastructureCoreConfiguration.cs (hunk 3 above)
Comment (raix) [reply application/shared-kernel/InfrastructureCore/InfrastructureCoreConfiguration.cs:14]: It should, it's part of the MSBuild spec

File: application/account-management/Api/Api.csproj (hunk 4 above)
Comment (raix) [reply application/account-management/Api/Api.csproj]: It's accordingly to the [NSwag documentation](https://github.com/RicoSuter/NSwag/wiki/NSwag.MSBuild) - Could be a typo - I'll remove it, only reason could be related to the build information in the assembly but let's see :)

**Pull request merged**